### PR TITLE
Beam-2171 power function 

### PR DESCRIPTION
**Pull request by app-tarush**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by XuMingmin** [comment]
R: @XuMingmin @xumingming 
CC: @jbonofre

**Comment by XuMingmin** [comment]
retest this please

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionBinaryExpression.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  private BeamSqlFunctionBinaryExpression(List<BeamSqlExpression> operands,
+      SqlTypeName outputType) {
+    super(operands, outputType);
+  }
+
+  public BeamSqlFunctionBinaryExpression(List<BeamSqlExpression> operands) {
+    this(operands, SqlTypeName.ANY);
+  }
+
+  @Override public boolean accept() {
```
Comment (xumingming) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionBinaryExpression.java]: The logic of  `accept`(only numeric types are permitted) does not name of the class `BeamSqlFunctionBinaryExpression`.

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionBinaryExpression.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    boolean acceptance = false;
+    if (operands.size() != 2) {
+      return false;
```
Comment (xumingming) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionBinaryExpression.java]: a minor comment: sometimes it just `return false`, sometimes a flag is set `line 32`, keep consistent might be better :)

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionBinaryExpression.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    if (SqlTypeName.NUMERIC_TYPES.contains(operands.get(0).getOutputType())
```
Comment (xumingming) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionBinaryExpression.java]: there is some convenient methods in `BeamSqlExpression` you might find handy: `opType`, `op`, `opValueEvaluated`

File: dsls/sql/src/test/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionUnaryExpression.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package org.apache.beam.dsls.sql.interpreter.operator.functions;
+
+/**
+ * Created by tarushgrover on 11/05/17.
+ */
+public class BeamSqlFunctionUnaryExpression {
```
Comment (xumingming) [inline dsls/sql/src/test/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionUnaryExpression.java]: this need to be removed?

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionUnaryExpression.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    if (SqlTypeName.NUMERIC_TYPES.contains(operands.get(0).getOutputType())) {
+      acceptance = true;
+    }
+    return acceptance;
+  }
+
+  /**
+   * For the operands of type {@link SqlTypeName#INT_TYPES}.
+   * */
+  public abstract Long calculate(Long leftOp, Long rightOp);
```
Comment (xumingming) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionUnaryExpression.java]: unary functions should have only one param?

File: dsls/sql/src/test/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionUnaryExpression.java [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package org.apache.beam.dsls.sql.interpreter.operator.functions;
```
Comment (XuMingmin) [inline dsls/sql/src/test/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionUnaryExpression.java]: Is this class duplicated?

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionBinaryExpression.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+import java.util.List;
+import org.apache.beam.dsls.sql.interpreter.operator.BeamSqlExpression;
+import org.apache.beam.dsls.sql.interpreter.operator.BeamSqlPrimitive;
+import org.apache.beam.dsls.sql.schema.BeamSQLRow;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+
+/**
+ * Base class for all binary functions.
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionBinaryExpression.java]: add a comment to list all binary functions here?

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionUnaryExpression.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
+package org.apache.beam.dsls.sql.interpreter.operator.functions;
+
+import java.util.List;
+import org.apache.beam.dsls.sql.interpreter.operator.BeamSqlExpression;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+
+/**
+ * Base class for all unary functions.
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionUnaryExpression.java]: please add a comment to list all unary functions here

**Comment by XuMingmin** [comment]
retest this please

**Comment by jbonofre** [comment]
retest this please

**Comment by app-tarush** [comment]
I somehow need help that in my local tests are running fine but when I submit pull request it says that checks are failing

**Comment by XuMingmin** [comment]
@app-tarush The error points that 
```
Caused by: org.apache.rat.mp.RatCheckException: Too many files with unapproved license: 2 See RAT report in: /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_MavenInstall@2/target/beam-parent-0.7.0-SNAPSHOT.rat
```

Have a quick scan, no license are added for the two files:
```
BeamSqlFunctionBinaryExpression.java
BeamSqlFunctionBinaryExpressionTest.java
```

Btw, how did you do the test locally? I thought `mvn clean verify` can detect this issue.

**Comment by app-tarush** [comment]
After adding license I ran it locally it ran but when I am creating a new
pull request for the changes then it is failing.

Regards,
Tarush

**Comment by XuMingmin** [comment]
From latest code, I still don't see license information for the two classes, are they submitted?

**Pull request closed without merging**